### PR TITLE
Professional experiences index

### PR DESCRIPTION
**Pull request by nehamamandelbaum**
This PR solves issue #565, which was to create an index page for professional experiences in the user interface (outside of the admin interface). 

How the UI looks: 

Captura de Tela 2023-10-24 às 08 36 25

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -89,9 +91,7 @@
     end
   end
 
-  if Rails.env.development?
-    mount LetterOpenerWeb::Engine, at: "/letter_opener"
-  end
+  mount LetterOpenerWeb::Engine, at: "/letter_opener" if Rails.env.development?
```
Comment (carloslopes) [inline config/routes.rb:94]: 😍

**Review by carloslopes — APPROVED** [review]

File: app/views/professional_experiences/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      <div class="card-header">
+        <i class="fa fa-table"></i>
+        <%= t('activerecord.models.professinal_experience.other') %>
+      </div>
+      <div class="card-body">
+        <div class="table-responsive">
+          <table class="table table-hover table-bordered users-evaluation" id="dataTable">
+            <thead>
+              <tr>
+                <th><%= t('activerecord.attributes.professional_experience.company') %></th>
```
Comment (adbatista) [inline app/views/professional_experiences/index.html.erb:13]: you can use here human_attribute_name

https://api.rubyonrails.org/classes/ActiveModel/Translation.html#method-i-human_attribute_name

https://guides.rubyonrails.org/i18n.html#translations-for-active-record-models

File: app/views/professional_experiences/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+              <% if @professional_experiences.empty?  %>
+                <tr>
+                  <td colspan="4"><%= t('professional_experience.not_found') %></td>
+                </tr>
+              <% end %>
+              <% @professional_experiences.each do |experience| %>
+                <tr>
+                  <td><%= experience.company %></td>
+                  <td><%= experience.position %></td>
+                  <td><%= l(Date.parse(experience.start_date), format: :short) %></td>
```
Comment (adbatista) [inline app/views/professional_experiences/index.html.erb:30]: why the date parse, aren't start_date and end_date already a date?

File: app/models/professional_experience.rb [hunk 4]
```diff
@@ -20,13 +20,15 @@ def self.ordered_by_start_date
   def greater_than_current_month?
     return if start_date.to_date < Date.today.end_of_month
 
-    errors.add(:start_date, I18n.t('activerecord.errors.models.professional_experience.attributes.start_date.greater_than_current_month'))
+    errors.add(:start_date,
+               I18n.t('activerecord.errors.models.professional_experience.attributes.start_date.greater_than_current_month'))
```
Comment (adbatista) [inline app/models/professional_experience.rb:24]: you can remove this whole translation path and let the rails error class to resolve this

https://guides.rubyonrails.org/i18n.html#error-message-scopes

File: app/views/professional_experiences/index.html.erb (hunk 3 above)
Comment (nehamamandelbaum) [reply app/views/professional_experiences/index.html.erb:30]: It's saved as a string. Initially, I tried to parse and it threw and error saying that in can't format strings.

File: app/models/professional_experience.rb (hunk 4 above)
Comment (adbatista) [reply app/models/professional_experience.rb:24]: you can pass the error type as argument https://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-add

**Review by adbatista — APPROVED** [review]

**Pull request merged**